prov/psm2/src/psmx2.h

```diff
@@ -706,36 +660,39 @@ int	psmx2_fabric(struct fi_fabric_attr *attr,
 		    struct fid_fabric **fabric, void *context);
 int	psmx2_domain_open(struct fid_fabric *fabric, struct fi_info *info,
 			 struct fid_domain **domain, void *context);
-int	psmx2_wait_open(struct fid_fabric *fabric, struct fi_wait_attr *attr,
-		       struct fid_wait **waitset);
 int	psmx2_ep_open(struct fid_domain *domain, struct fi_info *info,
 		     struct fid_ep **ep, void *context);
 int	psmx2_stx_ctx(struct fid_domain *domain, struct fi_tx_attr *attr,
 		     struct fid_stx **stx, void *context);
 int	psmx2_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 		     struct fid_cq **cq, void *context);
-int	psmx2_eq_open(struct fid_fabric *fabric, struct fi_eq_attr *attr,
-		     struct fid_eq **eq, void *context);
 int	psmx2_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 		     struct fid_av **av, void *context);
 int	psmx2_cntr_open(struct fid_domain *domain, struct fi_cntr_attr *attr,
 		       struct fid_cntr **cntr, void *context);
-int	psmx2_poll_open(struct fid_domain *domain, struct fi_poll_attr *attr,
-		       struct fid_poll **pollset);
+int	psmx2_wait_open(struct fid_fabric *fabric, struct fi_wait_attr *attr,
+			struct fid_wait **waitset);
 
 static inline void psmx2_fabric_acquire(struct psmx2_fid_fabric *fabric)
 {
-	++fabric->refcnt;
+	atomic_inc(&fabric->util_fabric.ref);
 }
 
-void	psmx2_fabric_release(struct psmx2_fid_fabric *fabric);
+static inline void psmx2_fabric_release(struct psmx2_fid_fabric *fabric)
+{
+	atomic_dec(&fabric->util_fabric.ref);
+}
 
 static inline void psmx2_domain_acquire(struct psmx2_fid_domain *domain)
 {
-	++domain->refcnt;
+	atomic_inc(&domain->util_domain.ref);
+}
+
+static inline void psmx2_domain_release(struct psmx2_fid_domain *domain)
+{
+	atomic_dec(&domain->util_domain.ref);
 }
```

Comment:
They are needed for allowing the same fabric object be returned for multiple fi_fabric calls. I do see a bug on how these calls are used. Will update the PR.